### PR TITLE
Refactor LineAttachToVoltageLevel modification with new generic design

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
Please use english everywhere

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 1]
```diff
@@ -1883,12 +1803,17 @@ private List<ModificationInfos> execCreateLinesAttachToSplitLinesCreation(Networ
...
-        assertLineAttachToVoltageLevelInfosNotEmpty(lineAttachToVoltageLevelInfos);
+    @Transactional
+    public List<ModificationInfos> createLineAttachToVoltageLevelCreation(@NonNull UUID networkUuid, String variantId, @NonNull UUID groupUuid, @NonNull UUID reportUuid,
+                                                                          @NonNull String reporterId, @NonNull LineAttachToVoltageLevelInfos lineAttachToVoltageLevelInfos) {
         ModificationNetworkInfos networkInfos = getNetworkModificationInfos(networkUuid, variantId);
         NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkUuid, groupUuid, networkModificationRepository, equipmentInfosService, false, networkInfos.isApplyModifications());
-        return execCreateLineAttachToVoltageLevelCreation(listener, lineAttachToVoltageLevelInfos, reportUuid, reporterId);
+        return handleModification(lineAttachToVoltageLevelInfos, listener, groupUuid, reportUuid, reporterId).stream().map(ModificationInfos.class::cast).collect(Collectors.toList());
+    }
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1812]: yes

File: src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        } else {
+            voltageLevelId = modificationInfos.getExistingVoltageLevelId();
+        }
+
+        LineAdder lineAdder = network.newLine()
+                .setId(attachmentLineInfos.getEquipmentId())
+                .setName(attachmentLineInfos.getEquipmentName())
+                .setR(attachmentLineInfos.getSeriesResistance())
+                .setX(attachmentLineInfos.getSeriesReactance())
+                .setG1(attachmentLineInfos.getShuntConductance1() != null ? attachmentLineInfos.getShuntConductance1() : 0.0)
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java:59]: yes

File: src/main/java/org/gridsuite/modification/server/dto/LineAttachToVoltageLevelInfos.java [hunk 3]
```diff
@@ -57,4 +63,37 @@ public class LineAttachToVoltageLevelInfos extends ModificationInfos {
...
+    }
+
+    @Override
+    public NetworkModificationException.Type getErrorType() {
+        return NetworkModificationException.Type.LINE_ATTACH_ERROR;
+    }
+
+    @Override
+    public Reporter createSubReporter(ReporterModel reporter) {
+        return reporter.createSubReporter("LineAttachToVoltageLevel", "Line attach to voltage level");
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/dto/LineAttachToVoltageLevelInfos.java:97]: yes

File: src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                null,
+                modificationInfos.getNewLine1Id(),
+                modificationInfos.getNewLine1Name(),
+                modificationInfos.getNewLine2Id(),
+                modificationInfos.getNewLine2Name(),
+                line,
+                lineAdder
+        );
+
+        algo.apply(network, false, subReporter);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java:85]: true

File: src/test/java/org/gridsuite/modification/server/LineAttachToVoltageLevelTest.java [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+        assertEquals(30., firstModification.getPercent(), 0.);
+        assertEquals("newLine1Id", firstModification.getNewLine1Id());
+        assertEquals("newLine2Id", firstModification.getNewLine2Id());
+        assertEquals("newLine1Name", firstModification.getNewLine1Name());
+        assertEquals("newLine2Name", firstModification.getNewLine2Name());
+        assertEquals("newAttPointName", firstModification.getAttachmentPointName());
+    }
+
+    @Test
+    public void testDeletion() throws Exception {
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/LineAttachToVoltageLevelTest.java:245]: testDelete ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java (hunk 4 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java:85]: ok

File: src/test/java/org/gridsuite/modification/server/LineAttachToVoltageLevelTest.java (hunk 5 above)
Comment (dbraquart) [reply src/test/java/org/gridsuite/modification/server/LineAttachToVoltageLevelTest.java:245]: ok

**Review by SlimaneAmar — APPROVED** [review]

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelWithNewVoltageLevelTest.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.UUID;
+
+import static org.gridsuite.modification.server.utils.MatcherLineAttachToVoltageLevelInfos.createMatcherLineAttachToVoltageLevelInfos;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+public class LineAttachToVoltageLevelWithNewVoltageLevelTest extends AbstractNetworkModificationTest {
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelWithNewVoltageLevelTest.java:23]: Rename by LineAttachToNewVoltageLevelTest ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+ */
+public class LineAttachToVoltageLevel extends AbstractModification {
+
+    private final LineAttachToVoltageLevelInfos modificationInfos;
+
+    public LineAttachToVoltageLevel(LineAttachToVoltageLevelInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    private Double zeroIfNull(Double d) {
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java:34]: yes done

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelWithNewVoltageLevelTest.java (hunk 6 above)
Comment (dbraquart) [reply src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelWithNewVoltageLevelTest.java:23]: I think we should keep <class>Test / <class>With<specialcase>Test.
it's easier to see we are testing the same class.
and it is consistent with LineAttachToVoltageLevelTest / LineAttachToVoltageLevelWithNewVoltageLevelTest

File: src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        if (attachmentLineInfos == null) {
+            throw new NetworkModificationException(LINE_ATTACH_ERROR, "Missing required attachment line description");
+        }
+
+        Line line = network.getLine(modificationInfos.getLineToAttachToId());
+        if (line == null) {
+            throw new NetworkModificationException(LINE_NOT_FOUND, modificationInfos.getLineToAttachToId());
+        }
+
+        String voltageLevelId;
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevel.java:50]: yes code has been changed with last PowSyBl. voltageLevelId is not used anymore.

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/LineAttachToVoltageLevelEntity.java [hunk 9]
```diff
@@ -85,6 +83,27 @@ public LineAttachToVoltageLevelEntity(String lineToAttachToId, double percent,
         this.newLine2Name = newLine2Name;
     }
 
+    @Override
+    public void update(ModificationInfos modificationInfos) {
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/modification/LineAttachToVoltageLevelEntity.java:87]: yes done

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**